src/stores/RoomViewStore.tsx

```diff
@@ -446,6 +446,33 @@ export class RoomViewStore extends Store<ActionPayload> {
                     description = _t("The person who invited you has already left, or their server is offline.");
                 }
             }
+
+            // if joined through room ID and didn't provide any via servers, provide
+            // a more detailed error than "No known servers"
+            if (roomId === this.state.roomId && !this.state.viaServers?.length) {
+                description = <div>
+                    { _t("This Room ID (%(id)s) cannot be joined. Try to join using a Room " +
+                         "Alias (#room:example.com) instead.", {
+                        id: this.state.roomId,
+                    }) }<br /><br />
+
+                    <details>
+                        <summary>{ _t("Why?") }</summary><br />
+
+                        { _t("A Room ID alone cannot be used to join a room unless your " +
+                             "homeserver already participates in that room.") }<br /><br />
+
+                        { _t("For a Room ID to be joinable over federation, a list of " +
+                             "homeservers to attempt to join through must be provided. " +
+                             "You didn't provide any.") }<br /><br />
+
+                        { _t("The sole purpose of the domain at the end of a Room ID " +
+                             "is to ensure it is unique.") }<br /><br />
+
+                        { _t("For these reasons, Room Aliases are the recommended way to join rooms.") }
+                    </details>
+                </div>;
+            }
```

Comment:
Related to https://github.com/vector-im/element-web/issues/24475

Maybe some language inspiration in this:

> Failed to join
> 
> The `!JsbthXjlaqosxISjgl:gitter.im` room isn't known to your homeserver (`matrix.org`).
> 
> You can try using a "via" server to join through which is supplied as the 2nd parameter in the `/join` slash command: `/join !JsbthXjlaqosxISjgl:gitter.im gitter.im`
> 
> Or if you're using a matrix.to link, you can try adding a `?via=gitter.im` query parameter to the URL: `https://matrix.to/#/!JsbthXjlaqosxISjgl:gitter.im?via=gitter.im`
>
> Alternatively, if you know a room alias (starts with a `#`) associated with this room, that will always be routable and allow you to join.
>
> Error details (expand/collapse)


![](https://user-images.githubusercontent.com/558581/217683868-e2fc4c85-40d6-4116-b29e-70b7d66536b8.png)

Reply:
1. I feel like using an alias is the more user friendly option and should be suggested first (move up to 2nd paragraph).
2. If unavailable, the next best option is to have whoever provided the alias provide a matrix.to with vias (automatically, by using their client's share or "copy room link" function).
3. Then the more advanced stuff about the join command can follow.

Reply:
Not all rooms have an alias and you can stumble upon a room ID in an unlimited number of ways. The user obviously tried to use an room ID here and we should make it possible for them to continue with what they have.

Leaving the user helpless to go ask someone else for a room alias isn't very empowering.

Reply:
I don't have any data to argue that point, except I think the user probably doesn't have that choice and can at best make a very educated guess.

However I do disagree with your previous suggestion:

> You can try using a "via" server to join through which is supplied as the 2nd parameter in the `/join` slash command: `/join !JsbthXjlaqosxISjgl:gitter.im gitter.im`

Honestly, either
- there is no way to progress at this point, because how are you supposed to guess a via server when the room ID does not hint at one (I think new IDs don't have them anymore?). Basically what I already said.
- it is completely ridiculous to ask users to repeat part of the room ID (`!room:domain.tld domain.tld`) in some command instead of detecting and trying it automatically.

The best I can come up with is to "try using the server (latter part of matrix ID) or a user who you know is in the room as a via server)".

Reply:
> (I think new IDs don't have them anymore?).

All room IDs include this information at the moment. And is why I created https://github.com/vector-im/element-web/issues/24475 to suggest to people how to do the right thing.

> it is completely ridiculous to ask users to repeat part of the room ID (!room:domain.tld domain.tld) in some command instead of detecting and trying it automatically.

Agreed that this should just be automatic, but I think we're at ideological odds with the decision makers since they would want to treat room IDs as opaque strings that shouldn't be parsed. Treating it as an opaque string is generally a good heuristic since there is some future-looking here where room IDs potentially won't include this information (wish I had some links for future desire).

The half-measure of suggesting the right thing at least gets them to the right place and teaches for next time. And I think has the potential to actually be merged.